packages/eslint-plugin/lib/config/core.ts

```diff
@@ -1,158 +1,144 @@
+import js from "@eslint/js";
 import type { Linter } from "eslint";
-import { sourceFiles } from "../utils/patterns";
+import { concat } from "eslint-flat-config-utils";
+import globals from "globals";
+import { javaScriptFiles } from "../utils/patterns.ts";
 
-const config: Linter.Config = {
-    overrides: [
-        {
-            files: sourceFiles,
-            plugins: ["import"],
-            extends: [
-                "eslint:recommended"
-            ],
-            parserOptions: {
-                sourceType: "module",
-                ecmaVersion: "latest"
-            },
-            env: {
-                es2024: true,
-                node: true,
-                commonjs: true
-            },
-            rules: {
-                // eslint:recommended overwrite some rules
-                "no-cond-assign": ["error", "except-parens"],
-                "no-labels": ["warn", { allowLoop: true, allowSwitch: false }],
-                "no-prototype-builtins": "off",
+const config: Linter.FlatConfig[] = await concat(
+    js.configs.recommended as Linter.FlatConfig,
```

Comment:
Is the "as Linter.FlatConfig" still required?

Reply:
Yes, because the type of `js.configs.recommended` is actually a subset of `Linter.FlatConfig` (it only contains the rules) and it confuses TypeScript as to what exactly should be in the array. I makes it mark everything that isn't `rules` in the second object as unknown keys. Since it's a subset, then casting it to `Linter.FlatConfig` clears everything up.